### ISSUE TITLE
[Bug]Failed to convert the model which is re-trained from Faster R-CNN ResNet50 V1 640x640 in TensorFlow 2 model zoo

### ISSUE THREAD

**Issue by dlod-openvino** [openvinotoolkit/openvino#5573]
##### System information (version)
- OpenVINO => 2021.3
- Operating System / Platform => Windows 64 Bit
- Sotfware Platform => Python 3.7, Tensorflow 2.4.1
- Problem classification: Model Conversion
- Framework: TensorFlow 2 Object Detection API, https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2.md
- Model name: Faster R-CNN ResNet50 V1 640x640 in TensorFlow 2 Model Zoo https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md

##### Detailed description
Failed to convert the model which is re-trained from Faster R-CNN ResNet50 V1 640x640
Get the Error information:

[ ERROR ]  -------------------------------------------------
[ ERROR ]  ----------------- INTERNAL ERROR ----------------
[ ERROR ]  Unexpected exception happened.
[ ERROR ]  Please contact Model Optimizer developers and forward the following information:
[ ERROR ]  Exception occurred during running replacer "ObjectDetectionAPIOutputReplacement (<class 'extensions.front.tf.ObjectDetectionAPI.ObjectDetectionAPIOutputReplacement'>)": 'inputs'
[ ERROR ]  Traceback (most recent call last):
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\utils\class_registration.py", line 290, in apply_transform
    for_graph_and_each_sub_graph_recursively(graph, replacer.find_and_replace_pattern)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\middle\pattern_match.py", line 60, in for_graph_and_each_sub_graph_recursively
    for_each_sub_graph_recursively(graph, func)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\middle\pattern_match.py", line 54, in for_each_sub_graph_recursively
    for_each_sub_graph(graph, recursive_helper)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\middle\pattern_match.py", line 39, in for_each_sub_graph
    func(node[sub_graph_name])
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\middle\pattern_match.py", line 50, in recursive_helper
    func(sub_graph)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\front\tf\replacement.py", line 48, in find_and_replace_pattern
    self.transform_graph(graph, desc._replacement_desc['custom_attributes'])
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\extensions\front\tf\ObjectDetectionAPI.py", line 1222, in transform_graph
    add_output_ops(graph, _outputs, graph.graph['inputs'])
KeyError: 'inputs'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\main.py", line 345, in main
    ret_code = driver(argv)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\main.py", line 309, in driver
    ret_res = emit_ir(prepare_ir(argv), argv)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\main.py", line 252, in prepare_ir
    graph = unified_pipeline(argv)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\pipeline\unified.py", line 29, in unified_pipeline
    class_registration.ClassType.BACK_REPLACER
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\utils\class_registration.py", line 340, in apply_replacements
    apply_replacements_list(graph, replacers_order)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\utils\class_registration.py", line 330, in apply_replacements_list
    num_transforms=len(replacers_order))
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\utils\logger.py", line 124, in wrapper
    function(*args, **kwargs)
  File "c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer\mo\utils\class_registration.py", line 318, in apply_transform
    )) from err
Exception: Exception occurred during running replacer "ObjectDetectionAPIOutputReplacement (<class 'extensions.front.tf.ObjectDetectionAPI.ObjectDetectionAPIOutputReplacement'>)": 'inputs'

[ ERROR ]  ---------------- END OF BUG REPORT --------------
[ ERROR ]  -------------------------------------------------

##### Steps to reproduce

**STEP1**. Create the TensorFlow 2 Object Detection API environment, refer to:  https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/

**STEP2**. Re-trained the model from Faster R-CNN ResNet50 V1 640x640 
Please download the labelled images and models from baidu disc:
link：https://pan.baidu.com/s/1XRPD6ba3ZI6nRCsxXPIzlg 
psw：c312 

**STEP3**. in the exported_model folder, there is a saved_model exported by the python script: exporter_main_v2.py, which can be used by object_detection_example_2.py to do the inference successfully.

**STEP4**. Use the command to convert the model
in the path: c:\Program Files (x86)\Intel\openvino_2021.3.394\deployment_tools\model_optimizer>
excute the command:
python mo_tf.py --saved_model_dir d:\tf_train\workspaces\cats_dogs\exported_model\my_model\saved_model  --transformations_config extensions\front\tf\faster_rcnn_support_api_v2.0.json --tensorflow_object_detection_api_pipeline_config d:\tf_train\workspaces\cats_dogs\exported_model\my_model\pipeline.config --reverse_input_channels --output_dir d:\tf_train\workspaces\cats_dogs\IR_model --data_type FP16
![bug2](https://user-images.githubusercontent.com/59691336/117557765-04b71100-b0a9-11eb-9c08-d9d937a895b7.png)
![bug1](https://user-images.githubusercontent.com/59691336/117557767-07196b00-b0a9-11eb-9a58-3aea11897b8d.png)

##### Issue submission checklist

 - [√] I report the issue, it's not a question
   
 - [√ ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ √] There is reproducer code and related data files: images, videos, models, etc.

**Comment by dlod-openvino** [comment]
![bug1](https://user-images.githubusercontent.com/59691336/117557837-e9003a80-b0a9-11eb-9403-16fb387d5a34.png)
![bug2](https://user-images.githubusercontent.com/59691336/117557839-edc4ee80-b0a9-11eb-8dea-f0c79eeeb7f0.png)

**Comment by Iffa-Meah** [comment]
Hi,

I'm not sure whether you had noticed this, but in order to convert your model to IR using the OpenVINO Model Optimizer, You need to freeze your model.

There are **three ways** to store non-frozen TensorFlow models and load them to the Model Optimizer:
**1. Checkpoint:**

In this case, a model consists of two files:

    **1.inference_graph.pb or inference_graph.pbtxt
       2.checkpoint_file.ckpt**
If you do not have an inference graph file, refer to Freezing Custom Models in Python section.

**2.MetaGraph:**

In this case, a model consists of three or four files stored in the same directory:

     **1.model_name.meta
        2.model_name.index
        3.model_name.data-00000-of-00001 (digit part may vary)
        4.checkpoint (optional)**

3.SavedModel format of TensorFlow 1.x and 2.x versions:

In this case, a model consists of a special directory with a **.pb file** and several subfolders: **variables, assets, and assets.extra**. 

Refer to this [official documentation]([url](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_TensorFlow.html)) for further info.

I had downloaded the model name: Faster R-CNN ResNet50 V1 640x640 in TensorFlow 2 Model Zoo https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md

and this model is insufficient to be used with model downloader right away since it is not frozen.

If you have the complete model, perhaps the one that you had frozen, please help to share.
Or else, you may try to use this frozen model: faster_rcnn_resnet50_coco which can be downloaded from OpenVINO model downloader.

**Comment by dlod-openvino** [comment]
@Iffa-Meah Thank you very much for your detailed explanation!~

**Comment by dlod-openvino** [comment]
@Iffa-Meah Thank you very much for your detailed explanation!

For TensorFlow 2.x versions, please ignore TF 1.x, the problem can be reproduced by 3 steps:

**Comment by dlod-openvino** [comment]
@Iffa-Meah Thank you very much for your detailed explanation!

For TensorFlow 2.x versions, please ignore TF 1.x, the problem can be reproduced by 3 steps:

1. downloaded the model name: Faster R-CNN ResNet50 V1 640x640 in TensorFlow 2 Model Zoo https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md
2. retrain the model, then get the checkpoint file by the models\research\object_detection\model_main_tf2.py
3. Using models\research\object_detection\exporter_main_v2.py to convert the checkpoint files to savedmodel
4. Using openvino 2021.3.394 to convert the savedmodel, you'll get the ERROR as below

The detailed steps for retraining the model is described by <[TensorFlow 2 Object Detection API tutorial](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/install.html) >

In conclusion: OpenVINO 2021.3.394 failed to convert the model retrained by [ TensorFlow 2 Object Detection API](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2.md)

**Comment by dlod-openvino** [comment]
BTW
[**Succeeded**] Convert the pre-trained model from TF2 Zoo (from https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md) by OpenVINO 2021.3.394

[**Failed**] Convert the exported model (from 'exporter_main_tf2.py'(from https://github.com/tensorflow/models/blob/master/research/object_detection/exporter_main_v2.py) after retraining by OpenVINO 2021.3.394

Hope to be verified this issue ASAP by any developers who are related MO. Thanks so much!

**Comment by Iffa-Meah** [comment]
Hi,

You are right, the original model can be converted to IR
![tf2_ir](https://user-images.githubusercontent.com/79643476/120157858-09915f80-c226-11eb-9f07-7ee21ab51d60.png)


Would really appreciate if you could share the re-trained model which you have issue with.

**Comment by Iffa-Meah** [comment]
Closing issue, feel free to re-open or start a new issue if additional assistance is needed.
